homeassistant/components/airzone_cloud/binary_sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Support for the Airzone Cloud binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Final
+
+from aioairzone_cloud.const import AZD_NAME, AZD_PROBLEMS, AZD_WARNINGS, AZD_ZONES
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+from .entity import AirzoneEntity, AirzoneZoneEntity
+
+
+@dataclass
+class AirzoneBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes Airzone Cloud binary sensor entities."""
+
+    attributes: dict[str, str] | None = None
+
+
+ZONE_BINARY_SENSOR_TYPES: Final[tuple[AirzoneBinarySensorEntityDescription, ...]] = (
+    AirzoneBinarySensorEntityDescription(
+        attributes={
+            "warnings": AZD_WARNINGS,
+        },
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        key=AZD_PROBLEMS,
+        name="Problem",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone Cloud binary sensors from a config_entry."""
+    coordinator: AirzoneUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors: list[AirzoneBinarySensor] = []
+
+    for zone_id, zone_data in coordinator.data.get(AZD_ZONES, {}).items():
+        for description in ZONE_BINARY_SENSOR_TYPES:
+            if description.key in zone_data:
+                binary_sensors.append(
+                    AirzoneZoneBinarySensor(
+                        coordinator,
+                        description,
+                        entry,
+                        zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(binary_sensors)
+
+
+class AirzoneBinarySensor(AirzoneEntity, BinarySensorEntity):
```

Comment:
Is this class needed, will there be other classes?

Reply:
Yes, there will be other classes, this is going to be very similar to Airzone (Local):
https://github.com/home-assistant/core/blob/31e9d95f66c3599ce56eceb258d0a48d7bfbb7a1/homeassistant/components/airzone/binary_sensor.py#L117
https://github.com/home-assistant/core/blob/31e9d95f66c3599ce56eceb258d0a48d7bfbb7a1/homeassistant/components/airzone/binary_sensor.py#L139
https://github.com/home-assistant/core/blob/31e9d95f66c3599ce56eceb258d0a48d7bfbb7a1/homeassistant/components/airzone/binary_sensor.py#L158